sui_core/src/gateway_state.rs

```diff
@@ -908,25 +634,24 @@ where
         gas_payment: ObjectID,
         gas_budget: u64,
     ) -> Result<TransactionData, anyhow::Error> {
-        let account = self.get_or_create_account(signer)?;
-        let primary_coin = account.latest_object_ref(&primary_coin)?;
-        let coin_to_merge = account.latest_object_ref(&coin_to_merge)?;
-        let gas_payment = account.latest_object_ref(&gas_payment)?;
+        // TODO: We should be passing in object_refs directly instead of object_ids.
```

Comment:
This comment is resolved now?

Reply:
No. We need to change the `merge_coins` interface to pass in ObjectRefs instead of ObjectIDs.